### ISSUE TITLE
Create a color-analyzer core function

### ISSUE THREAD

**Issue by masakudamatsu** [masakudamatsu/color-analyzer#1]
Input: a string of RGB css code

Outputs:
- `luminance`: Contrast ratio against pure black (1 to 21, up to 2 decimal digits)
- `saturation`: Share of pure hue (as oppose to neutral color) in percentage (0 to 100)
- `hue`: Pure hue (0 to 359) that is mixed with `neutralColor` to produce the input color (0 to 359)
- `neutralColor`: Neutral color that is mixed with `hue` to produce the input color (0 to 255)

**Comment by masakudamatsu** [comment]
To parse the input string, we use [`parse-color` package](https://www.npmjs.com/package/parse-color).
- Weekly downloads: 327,696 (as of 10 Dec, 2020)
- No issue pending
- [used by `get-relative-luminance `library](https://github.com/misund/get-relative-luminance/blob/master/src/index.js)

**Comment by masakudamatsu** [comment]
To convert a user-selected color into luminance, we use our own converter `getContrastRatio.js`.
- This function rounds the calculated value up to 2 decimal places.
- What's available out there uses the `Math.floor()`, which makes sense to check whether a given contrast ratio satisfies the WGAC criteria of 3:1 or 4.5:1.
- For our purposes, however, we need the actual level of luminance. We don't want to show the contrast ratio of, say, 4.449 as 4.44.

**Comment by masakudamatsu** [comment]
Our test code verifies the conversion by aggregating outputs back into the input. 

However, the rounding error is unavoidable. An error of plus/minus 1 is allowed.
